### PR TITLE
Instructions for using conda packages in different environments

### PR DESCRIPTION
**Pull request by rbeyer**
## Description
These instructions provide two different mechanisms for users to 'use' ISIS from a 'different' conda environment than the one they installed ISIS into.

## Related Issues
#615 
#3570

## Motivation and Context
Addresses user concerns about how to deal with the pinning of ISIS dependencies in its conda env.

## How Has This Been Tested?
Ensured these instructions worked on my system, stepping through each instruction.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/) **This URL is returning a 404!**
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: README.md
Comment (jlaura) [inline README.md]: Maybe a better way to locate this is to do: `conda activate <the_isis_env>` and then `echo $CONDA_PREFIX`.

File: README.md
Comment (jlaura) [inline README.md]: Here we probably want more description about the what is going on. For example, many people will have installations in `$HOME/anaconda3` or `$HOME/miniconda3`. For the copy/paste user this will cause issues.

File: README.md
Comment (jlaura) [inline README.md]: We should remove the 3 for all `isis` instances. This is a note for us too, to make the update to this README.

**Review by jlaura — CHANGES_REQUESTED** [review]

File: README.md
Comment (rbeyer) [reply README.md]: Ah, yes, good point.  I've added another step to do this.  Commit to PR forthcoming.

File: README.md
Comment (rbeyer) [reply README.md]: D'oh!  Right, I put anaconda in a leading-dot directory because I didn't want to see it all the time, I forgot that wasn't standard.  Fixed, and tried to add some extra explanation.

File: README.md
Comment (rbeyer) [reply README.md]: So I went through and cleared the 'isis3' string where it indicated the environment name in the conda instructions even above my new section.  There are still lots of other 'isis3' strings in this README, but they require changes in other places/files.

File: README.md
Comment (jlaura) [inline README.md:193]: :+1: This looks really good for the uninitiated in how Conda is storing envs.

**Review by jlaura — DISMISSED** [review]

**Comment by jlaura** [comment]
@thareUSGS @michaelaye @AndrewAnnex  You are users that I know also grapple with ISIS and other conda envs. Can you also review and/or test these instructions? 

A +1 response would be ideal before we merge.

File: README.md
Comment (rbeyer) [reply README.md:193]: Thanks.  I was new to conda just a year ago, and honestly, 24h ago, I didn't even know about the --stack option.

**Comment by michaelaye** [comment]
I'm actually right now setting up ISIS into a new system so it's ideal timing. will report back.

**Comment by AndrewAnnex** [comment]
I’ll take a look later tonight!

-Andrew Annex

**Comment by krlberry** [comment]
Thank you for the ISIS3 to ISIS cleanup in these docs!

**Comment by AndrewAnnex** [comment]
so I never knew about the stacking feature in conda, very neat. I've played around with the activation scripts before (actually for ASP @rbeyer so I can have multiple versions to switch between) so I am confident those instructions are correct.

But would it be possible to add a python entrypoint/command to isis to automate the procedure for deploying the activation/deactivation scripts so there can be a one liner command from the isis environment ie `copy_isis_activation other-conda-env`? This could be another pr obviously, and it is not clear how robust of a solution that would be compared to just having the well written instructions.

**Comment by michaelaye** [comment]
> But would it be possible to add a python entrypoint/command to isis to automate the procedure for deploying the activation/deactivation scripts so there can be a one liner command from the isis environment ie copy_isis_activation other-conda-env? This could be another pr obviously, and it is not clear how robust of a solution that would be compared to just having the well written instructions.

i think that's basically what I suggested [here](https://github.com/USGS-Astrogeology/ISIS3/issues/615#issuecomment-610700255), to boost the capability of isis3VarInit.py to do exactly that.
If people agree, I could work on that PR.

File: README.md
Comment (michaelaye) [inline README.md]: package (can't edit the file, it's grayed out, otherwise I'd have done it. Still don't fully understand the GitHub interface rules for this).

File: README.md
Comment (michaelaye) [inline README.md]: packages

File: README.md
Comment (michaelaye) [inline README.md:146]: the danger i smell here is, what happens when I do a "conda update --all" while stacked? Will it update (and maybe break) the isis env?

File: README.md
Comment (rbeyer) [reply README.md]: No worries, good for folks to fix their own PRs.  Commit incoming.

File: README.md
Comment (rbeyer) [reply README.md]: Double-typo: fixed.

File: README.md
Comment (rbeyer) [reply README.md:146]: I'm not sure, but that's a bit beyond the scope of these instructions.

I suspect that it will only affect the current environment, i.e. the one on the top of the 'stack,' and will not affect the isis environment that it is stacked on top of, but I haven't tested.

**Comment by jlaura** [comment]
@AndrewAnnex @michaelaye I think putting in a PR with the proposed env capabilities would absolutely be a nice to have.

**Comment by jlaura** [comment]
@AndrewAnnex @michaelaye Are you all able to review and approve this?

**Review by AndrewAnnex — DISMISSED** [review]
I think this is good as is

**Comment by michaelaye** [comment]
give me a few more hours and then i can try it out myself.

**Comment by michaelaye** [comment]
what's up with Jenkins?

**Comment by jlaura** [comment]
@michaelaye We are still working on getting Jenkins setup on OSX and to determine if/how to make the CI logs publicly available. We website can be hit from the outside world, but only we have the ability to log in right now. 

Basically, ignore Jenkins. We are working on it.

**Review by jlaura — DISMISSED** [review]

**Comment by michaelaye** [comment]
We should probably mention that the stacking method doesn’t work for Jupyter? At least I haven’t found a method how to do it. Otherwise this is good to go.

**Comment by michaelaye** [comment]
It could work but only in one special way: launching the Jupyter server within the stacked environment.

**Comment by jlaura** [comment]
@michaelaye Is this because of how ISIS is working or because of how Jupyter and stacking are working? If the former, we should definitely document. If the latter, it might be best to hold off from cluttering the ISIS docs until we get issues opened about this. Thoughts?

File: README.md
Comment (jessemapel) [inline README.md]: `#The-ISIS3-Data-Area` is a dead link now. The proper link is `#The-ISIS-Data-Area`

File: README.md
Comment (jessemapel) [reply README.md:146]: Just tested, `conda update --all` only updates the packages in the current environment, that would be working in this example.

File: README.md
Comment (jessemapel) [inline README.md:207]: You can actually just call the isis3VarInit.py script from your ISIS environment while you have your other environment active and will write the values into your current active environment. I'm not sure if that's more or less clear though.

File: README.md
Comment (jessemapel) [inline README.md]: I would emphasize that it's VERY important you add `$ISISROOT/bin` to the end of your path so that executables in your current environment hide executables in the ISIS environment.

**Review by jessemapel — COMMENTED** [review]
Looks good in general. The broken link needs to be fixed, but the other comments are just suggestions.

**Comment by jessemapel** [comment]
@michaelaye @jlaura I would rather see this merged and then a new PR/issue exploring how to make this work with Jupyter.

File: README.md
Comment (michaelaye) [reply README.md]: in danger of correcting too many things at once, what about the ISIS3DATA env variable name? Guess it's used SOO widely, no chance to change that anytime soon without breaking everything, right?

**Comment by michaelaye** [comment]
@jessemapel That sounds fair.

**Review by michaelaye — DISMISSED** [review]

File: README.md
Comment (rbeyer) [reply README.md]: @jessemapel, link fixed in upcoming commit.

@michaelaye, dude, what to name environment variables in the software suite is out-of-scope for a PR on changes to the README!

File: README.md
Comment (rbeyer) [reply README.md:207]: You could, but you'd still have to fuss with the paths manually, so you'd still have to go into a discussion of the paths, etc.  I'd have to add at least another paragraph here, and I already think this is too long.  Let me know if you think that's needed.

**Comment by jlaura** [comment]
@jessemapel Agreed. I suspect it doesn't belong in here at all.

File: README.md
Comment (rbeyer) [reply README.md]: Added to new commit.

**Comment by michaelaye** [comment]
> @jessemapel Agreed. I suspect it doesn't belong in here at all.

with the caveat that one wouldn't want to advertise features that don't work in X % (with X quite larger than 50) of people's work environments.

**Comment by jessemapel** [comment]
> with the caveat that one wouldn't want to advertise features that don't work in X % (with X quite larger than 50) of people's work environments.

I think it's reasonable to put a warning in if an approach doesn't work with Jupyter.

**Review by jessemapel — APPROVED** [review]
Looks good to me.

**Comment by thareUSGS** [comment]
I finally had a chance to test the new instructions (Ubuntu 18 under Windows WSL). Just some initial notes. This is not worth opening a ticket unless someone else finds the same.

(1) conda install -c usgs-astrogeology isis
Installs isis3.10.0 (not 4 as implied by the current README). Until v4 is official, should that be changed -- under section "ISIS Installation with Conda" step 6.

but more importantly (at least for WSL), for isis3.10.0 gives me an opencv error for GUIs.
```
qview $ISIS3DATA/base/dems/MSGR_DEM_USG_EQ_I_V02_prep.cub
qview: error while loading shared libraries: libopencv_calib3d.so.3.2: cannot open shared object file: No such file or directory
```

(2) This could be a WSL quirk. I could fix this by downgrading to v3.9.1 by running
conda install -c usgs-astrogeology isis3

**Comment by jlaura** [comment]
@thareUSGS I think we are going to sign you up to do the windows support.... 😸 

https://github.com/USGS-Astrogeology/ISIS3/wiki/FAQ#how-do-i-install-isis-on-windows

**Comment by thareUSGS** [comment]
Can do. There have always been plenty of reasons for us to support Windows, besides lots of our users who are stuck on it, but being able to better integrate ISIS workflows into Windows-only applications would be extremely beneficial (e.g. Socet SET/GXP, ArcMap, Global Mapper). Anyway, we might find that when future versions of Windows ship with a full Linux kernel (as done with WSL2 instead of an emulated env. as done with WSL1), Windows might end up being easier to support than Mac. :-). https://www.howtogeek.com/413564/windows-10-is-getting-a-built-in-linux-kernel/

**Pull request merged**
